app/pages/projects/[id].tsx

```diff
@@ -1,13 +1,16 @@
 import { useRouter } from 'next/router'
 import { ProjectForm } from '../../frontend/components/projectForm/projectForm'
 import { ProtectedPage } from '../../frontend/components/protectedPage'
-import { ProjectInput, useProjectsQuery, useProjectUpdateMutation } from '../../frontend/generated/graphql'
+import { ProjectInput, useProjectQuery, useProjectUpdateMutation } from '../../frontend/generated/graphql'
 
 const ProjectDetails = (): JSX.Element => {
-  const [{ data, fetching }] = useProjectsQuery()
   const router = useRouter()
   const { id } = router.query
-  const selectedProject = data?.projects.find((p) => p.id === id)
+  if (!id || !Number.parseInt(id.toString())) {
+    return <div>Loading...</div>
+  }
+  const [{ data, fetching }] = useProjectQuery({ variables: { projectId: Number.parseInt(id.toString()) } })
```

Comment:
That is a violation of the first "Rule(s) of Hooks": https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
Do you have a pattern to avoid calling useProjectQuery if id not yet read from next router? Mine would be useEffect ;)

Reply:
changed but problem remains.